service-providers/duetify-service.js

```diff
@@ -1,5 +1,58 @@
 'use strict';
+const qs = require('qs');
+const fetch = require('node-fetch');
+const FormData = require('form-data');
+const request = require('request-promise');
 
-module.exports.clientCallback = ctx => {
-  ctx.body = 'Client loads app here';
+const stateKey = 'spotify_auth_state';
+const redirectUri = 'http://localhost:8080/';
+const tokenUrl = 'https://accounts.spotify.com/api/token';
+
+
+module.exports.clientCallback = async (ctx, next) => {
+
+  const code = ctx.request.query.code || false;
+  const state = ctx.request.query.state || false;
+  const storedState = ctx.cookies.get(stateKey);
+
+  if (state === false || state !== storedState) {
+    ctx.body = 'State error';
+    ctx.status = 401;
+  } else {
+    ctx.cookies.set(stateKey, state, {expires: new Date(Date.now() - 10)});
+    const httpHeaders = {
+      'Authorization': 'Basic ' + (new Buffer(process.env.CLIENT_ID + ':' + process.env.CLIENT_SECRET).toString('base64'))
```

Comment:
Instead of `process.env` use `dotenv` you've installed it in your package already.